src/gstreamer/gstkvssink.cpp

```diff
@@ -1231,7 +1228,6 @@ put_frame(std::shared_ptr<KvsSinkCustomData> data, void *frame_data, size_t len,
             kvs_sink_metric->client_metrics = data->kinesis_video_producer->getMetrics();
             kvs_sink_metric->frame_pts = frame.presentationTs;
             kvs_sink_metric->on_first_frame = data->on_first_frame;
-            data->on_first_frame = false;
```

Comment:
Why are we removing this? In a weird situation where every frame is a key frame, this loop with run infinitely.

Reply:
Didn't intend to remove it, missed it while updating that section. Added it back. Thanks for pointing it out